next/components/ui/UserProfile/UserProfileView.tsx

```diff
@@ -56,16 +57,18 @@ const UserProfileView = () => {
       updateUserData(temporaryUserData).then(() => {
         setIsEditing(false)
         setIsAlertOpened(true)
+        setTimeout(() => setIsAlertOpened(false), 3000)
```

Comment:
would prefer to have all of `handleOnSubmitEditing` as async instead of having part of it in `then` callback. You also want to handle the catch part - show an error when the cognito write fails (and useful info if you can determine why it failed)